### ISSUE TITLE
HTTP API to kick out connections.

### ISSUE THREAD

**Issue by zsoltpardi** [simple-rtmp-server/srs#429]
Using the http api such as http://192.168.1.102:1985/api/v1/streams, when the stream query is issued to the http server the returned information includes a field for the bytes delivered, but the value is always 0. It would be very useful to get the delivered streamed bytes of the stream when it is queried via the http api. What I am asking is similar what is returned on the http callback when the player close the stream, then the delivered bytes are there which is again quite useful to know the amount of data was delivered to a particular player.

**Comment by winlinvip** [comment]
The bytes of api/v1/streams should not be 0, for example, http://ossrs.net:19851/api/v1/streams
```

{
id: 20790,
name: "demo",
vhost: 20781,
app: "live",
clients: 0,
send_bytes: 13020,
recv_bytes: 10980,
live_ms: 1434589989648,
status: "idling",
video: null,
audio: null
```

**Comment by winlinvip** [comment]
The bytes only stat the RTMP bytes, do you need SRS to stat the HLS/HTTP-FLV stream bytes?

**Comment by zsoltpardi** [comment]
No, that's fine to have this query only for RTMP streams and you are absolutely right, the bytes are there ... I am not sure why the send_bytes and rcv_bytes were 0 before for me, I have tested it again and indeed it works correctly. Thanks very much for replying.
On an other note, can I ask, is it possible to close/delete a stream using the http api or is the stream there forever once it was created? Or is there any other way to delete a stream from the server?

**Comment by zsoltpardi** [comment]
My question about closing the stream is mainly related to use cases when the client connection (that watch the stream) is opened with authentication token. The scenario is, our application allows client access using the http hook and token, and then after a certain time or upon certain events we need to close that client connection, i.e. kick out the client from the server. If it is not possible to delete the stream is there any way to close the client connection?

**Comment by winlinvip** [comment]
It's possible to kick out the specified client, which is not support by SRS yet.
U need two http apis:

1. To query all clients, or query specified client by some information. We can got the id of client to kick out.
1. To kick out the client by specified client id.

**Comment by zsoltpardi** [comment]
I am using the http apis, but I am not sure  how to perform step 1 and 2. Could you clarify please, are you saying it is not supported yet and you need to design this feature or is the feature there and the client can be terminated using step 1 and 2? I tried all http api calls but none of them seems can be used for this.

**Comment by winlinvip** [comment]
SRS2 not support these two apis to kick out connections.
I will support in SRS3+. So u can write code if u can, for SRS3+ is future things.

**Comment by misiek08** [comment]
It's implemented now, in 2.0 development version.
